client-js/app/elements/labrad-registry.ts

```diff
@@ -1,6 +1,15 @@
 import {RegistryApi, RegistryListing} from '../scripts/registry';
 import {Places} from '../scripts/places';
 
+type ListItem = {
+  name: string,
+  isParent: boolean,
+  isDir: boolean,
+  isKey: boolean,
+  url?: string,
+  value?: string,
+};
```

Comment:
Per the typescript language spec: "Interface types have many similarities to type aliases for object type literals, but since interface types offer more capabilities they are generally preferred to type aliases." (https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#3.10). Not a big deal in this case though.